### PR TITLE
[Enhancement] Profile optimization(1)

### PR DESCRIPTION
**Pull request by liuyehcf**
## Main works of this pr

Attach all the variables that differ from its default value to profile, in json format.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 38 (65.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[303, 322]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|21|23|91.30%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 38 (65.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[303, 322]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|21|23|91.30%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 38 (65.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[303, 322]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|21|23|91.30%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 38 (65.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[303, 322]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|21|23|91.30%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 38 (65.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[303, 322]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|21|23|91.30%|[2587, 2588]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 39 / 52 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|1|00.00%|[148]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[302, 321]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|0|2|00.00%|[233, 241]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadTask.java|0|6|00.00%|[1067, 1068, 1069, 1070, 1071, 1079]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|25|27|92.59%|[2592, 2593]|
|:large_blue_circle:|com/starrocks/common/util/RuntimeProfile.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ExplainAnalyzer.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|3|3|100.00%|[]|

**Review by satanson — APPROVED** [review]

File: test/sql/test_profile/R/test_profile [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+-- result:
+0
+Analyze profile succeeded
+Analyze profile succeeded
+Analyze profile succeeded
+-- !result
+shell: env mysql_cmd="${mysql_cmd} -D${db[0]}" bash ${root_path}/sql/test_profile/T/test_profile_non_default_variables.sh
+-- result:
+0
+NonDefaultSessionVariables contains 'pipeline_dop'
```
Comment (fzhedu) [inline test/sql/test_profile/R/test_profile:16]: why not show its current value? for example  16(default) ->  2(now)

Comment (liuyehcf) [reply test/sql/test_profile/R/test_profile:16]: this is the output of the shell script, the output of the profile will be checked at the script.

```sh
function test_non_default_variables() {
    analyze_output=$(${mysql_cmd} -e "$1")
    if grep -q "pipeline_dop: 0 -> 2" <<< "${analyze_output}"; then
        echo "NonDefaultSessionVariables contains 'pipeline_dop'"
    else
        echo "NonDefaultSessionVariables not contains 'pipeline_dop'"
        exit 1
    fi
}
```

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**